### PR TITLE
IGNITE-23760 Update Apache Ignite 2.17 release notes

### PR DESCRIPTION
**Pull request by NSAmelchev**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: RELEASE_NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,174 @@
 Apache Ignite Release Notes
 ===========================
 
+Apache Ignite In-Memory Distributed Database 2.17.0
+-----------------------------------------------------------
+
+(!) WARNINGS:
+* Control utility now uses thin client protocol by default (configured on a node via ClientConnectorConfiguration). Connection via REST protocol (ConnectorConfiguration) was deprecated.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt]: via Binary-REST

File: RELEASE_NOTES.txt [hunk 2]
```diff
@@ -1,6 +1,174 @@
...
+* Removed deprecated authorization methods from Security Context.
+* Removed deprecated TcpCommunicationSpi shared memory port configuration.
+
+Ignite:
+* Added 'keepRaw' parameter to DumpReaderConfiguration.
+* Added IndexQuery events.
+* Added TLSv1_3 to o.a.i.client.SslProtocol.
+* Added ability to create user defined SQL views.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:16]: To "SQL" or/and "SQL Calcite" section

File: RELEASE_NOTES.txt [hunk 3]
```diff
@@ -1,6 +1,174 @@
...
+* Added check for SSL errors on TcpDiscovery writing failures.
+* Added configuration for distributed properties default values.
+* Added expiry policy info to the caches system view.
+* Added libs to the OpenCensus module to make Prometheus metrics work from the box without building any extra libs.
+* Added logging for CDC mode switching.
+* Added operation time metrics for putAllConflict, removeAllConflict operations.
+* Added proper logging of default pageSize.
+* Added query thread pool metrics to log file.
+* Added snapshot creation time into the snapshot view.
+* Added starting square bracket in default log4j2 pattern.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:27]: Added usually means some new feature, here "added" looks like "fixed" some typo

File: RELEASE_NOTES.txt [hunk 4]
```diff
@@ -1,6 +1,174 @@
...
+* Added support of OR operator with dynamic parameters for index scans.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:28]: To `SQL Calcite`

File: RELEASE_NOTES.txt [hunk 5]
```diff
@@ -1,6 +1,174 @@
...
+* Fixed fields count serialization in TcpCommunication messages.
+* Fixed incremental snapshot restore fail if consistent ID is not a string.
+* Fixed integer overflow in page eviction trackers.
+* Fixed iterator for local IndexQuery.
+* Fixed lambda serialization issues in code deployment for Java 21.
+* Fixed local class deployments cache cleanup on reconnect.
+* Fixed logging of reduced cache entries for distributed long running operations.
+* Fixed long JVM pause detector.
+* Fixed node crashing if a snapshot restore was canceled due to network issues.
+* Fixed node fails while checking security permission for activation during join.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:54]: `node failure`?

File: RELEASE_NOTES.txt [hunk 6]
```diff
@@ -1,6 +1,174 @@
...
+* Added ability to create user defined SQL views.
+* Added address information to ClientConnectionException.
+* Added check for SSL errors on TcpDiscovery writing failures.
+* Added configuration for distributed properties default values.
+* Added expiry policy info to the caches system view.
+* Added libs to the OpenCensus module to make Prometheus metrics work from the box without building any extra libs.
+* Added logging for CDC mode switching.
+* Added operation time metrics for putAllConflict, removeAllConflict operations.
+* Added proper logging of default pageSize.
+* Added query thread pool metrics to log file.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:25]: `to the log file`?

File: RELEASE_NOTES.txt [hunk 7]
```diff
@@ -1,6 +1,174 @@
...
+* Fixed sending a stack trace to a thin client if task fails in map phase.
+* Fixed server node fail when node service filter class is unknown.
+* Fixed service descriptor when deployment failed.
+* Fixed warnings about closed socket.
+* Fixed wrong old index size in the defragmentation procedure.
+* Forbid a non-transactional IgniteCache#clear() inside a transaction.
+* Implemented custom metrics.
+* Implemented transaction aware SQL and Scan queries.
+* Improved check of previous node on new incoming server connection.
+* Moved Ignite sources to Java 11.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:71]: I think it also should be in WARNINGS section

File: RELEASE_NOTES.txt [hunk 8]
```diff
@@ -1,6 +1,174 @@
...
+* Fixed pending tree redundant cleanup on entries expiration.
+* Fixed redundant lambda object allocation in ClockPageReplacementFlags.
+* Fixed removal of required affinity assignment instance from affinity cache.
+* Fixed sending a stack trace to a thin client if task fails in map phase.
+* Fixed server node fail when node service filter class is unknown.
+* Fixed service descriptor when deployment failed.
+* Fixed warnings about closed socket.
+* Fixed wrong old index size in the defragmentation procedure.
+* Forbid a non-transactional IgniteCache#clear() inside a transaction.
+* Implemented custom metrics.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:68]: `custom user defined`?

File: RELEASE_NOTES.txt [hunk 9]
```diff
@@ -1,6 +1,174 @@
...
+* Updated jetty dependency version to 9.4.56.v20240826.
+* Updated json-path dependency version to 2.9.0.
+* Updated jtidy dependency version to 1.0.5.
+* Updated log4j2 dependency version to 2.22.0.
+* Updated opencensus dependency version to 0.31.1.
+* Updated snappy dependency version to 1.1.10.7.
+* Updated zookeeper dependency version to 3.8.4.
+
+SQL:
+* Added logging of the IdleVerify command.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:90]: Is it really related to SQL?

File: RELEASE_NOTES.txt [hunk 10]
```diff
@@ -1,6 +1,174 @@
...
+* Fixed rehashing of inputs containing null values.
+* Fixed rounding of numerics (now rounds all types of floating point numerics to nearest integer).
+* Fixed sensitive information removal failure for JOINs and other SQL nodes.
+* Updated Calcite version to 1.37.
+
+Control utility:
+* Added a flag to ignore existed caches to the cache create command.
+* Added ability to view cache data in JSON format.
+* Added histogram bounds to the metric command output.
+* Added support for the thin client connection to the control utility and deprecate legacy REST connection.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:143]: Binary-REST

File: RELEASE_NOTES.txt [hunk 11]
```diff
@@ -1,6 +1,174 @@
...
+* Fixed partition awareness with node filter.
+* Fixed stale client cluster group data after cluster restart.
+
+Ignite .NET:
+* Added compute task sessions support.
+* Fixed platform cache update for negative cache ID.
+* Improved thin client service invocation performance by sending requests directly to the nodes where the service is deployed.
+
+JDBC/ODBC:
+* Added support for stream APIs for BLOB.
```
Comment (alex-plekhanov) [inline RELEASE_NOTES.txt:170]: Why only stream API? AFAIK BLOB was not supported at all by JDBC

**Review by alex-plekhanov — APPROVED** [review]

File: RELEASE_NOTES.txt [hunk 12]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed node failure while checking security permission for activation during join.
+* Fixed node join failure when index creation is in progress and caches have the same deploymentId.
+* Fixed pending tree redundant cleanup on entries expiration.
+* Fixed redundant lambda object allocation in ClockPageReplacementFlags.
+* Fixed removal of required affinity assignment instance from affinity cache.
+* Fixed sending a stack trace to a thin client if task fails in map phase.
+* Fixed server node fail when node service filter class is unknown.
+* Fixed service descriptor when deployment failed.
+* Fixed starting square bracket in default log4j2 pattern.
+* Fixed warnings about closed socket.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:64]:
```suggestion
* Fixed warnings about a closed socket.
```

File: RELEASE_NOTES.txt [hunk 13]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed wrong old index size in the defragmentation procedure.
+* Forbid a non-transactional IgniteCache#clear() inside a transaction.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:66]:
```suggestion
* Forbade a non-transactional `IgniteCache#clear()` inside a transaction.
```

File: RELEASE_NOTES.txt [hunk 14]
```diff
@@ -1,6 +1,177 @@
...
+* Implemented custom user defined metrics.
+* Implemented transaction aware SQL and Scan queries.
+* Improved check of previous node on new incoming server connection.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:69]:
```suggestion
* Improved check of previous node on a new incoming server connection.
```

File: RELEASE_NOTES.txt [hunk 15]
```diff
@@ -1,6 +1,177 @@
...
+* Moved Ignite sources to Java 11.
+* Optimized UUID serialization for communication messages.
+* Optimized entries expiration for persistent caches.
+* Removed deprecated TcpCommunicationSpi shared memory port configuration.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:73]:
```suggestion
* Removed deprecated `TCPCommunicationSPI` shared memory port configuration.
```

File: RELEASE_NOTES.txt [hunk 16]
```diff
@@ -1,6 +1,177 @@
...
+* Removed deprecated authorization methods from Security Context.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:74]: Duplicates line 10

File: RELEASE_NOTES.txt [hunk 17]
```diff
@@ -1,6 +1,177 @@
...
+* Updated PostgreSQL JDBC dependency version to 42.7.3.
+* Updated Spring dependency version to 5.3.39.
+* Updated commons-io dependency version to 2.17.0.
+* Updated grpc-context version to 1.62.2.
+* Updated janino dependency version to 3.1.12.
+* Updated jetty dependency version to 9.4.56.v20240826.
+* Updated json-path dependency version to 2.9.0.
+* Updated jtidy dependency version to 1.0.5.
+* Updated log4j2 dependency version to 2.22.0.
+* Updated opencensus dependency version to 0.31.1.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:86]:
```suggestion
* Updated OpenCensus dependency version to 0.31.1.
```

File: RELEASE_NOTES.txt [hunk 18]
```diff
@@ -1,6 +1,177 @@
...
+* Updated snappy dependency version to 1.1.10.7.
+* Updated zookeeper dependency version to 3.8.4.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:88]:
```suggestion
* Updated ZooKeeper dependency version to 3.8.4.
```

File: RELEASE_NOTES.txt [hunk 19]
```diff
@@ -1,6 +1,177 @@
...
+* Updated Spring dependency version to 5.3.39.
+* Updated commons-io dependency version to 2.17.0.
+* Updated grpc-context version to 1.62.2.
+* Updated janino dependency version to 3.1.12.
+* Updated jetty dependency version to 9.4.56.v20240826.
+* Updated json-path dependency version to 2.9.0.
+* Updated jtidy dependency version to 1.0.5.
+* Updated log4j2 dependency version to 2.22.0.
+* Updated opencensus dependency version to 0.31.1.
+* Updated snappy dependency version to 1.1.10.7.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:87]:
```suggestion
* Updated Snappy dependency version to 1.1.10.7.
```

File: RELEASE_NOTES.txt [hunk 20]
```diff
@@ -1,6 +1,177 @@
...
+* Updated Netty dependency version to 4.1.101.Final.
+* Updated PostgreSQL JDBC dependency version to 42.7.3.
+* Updated Spring dependency version to 5.3.39.
+* Updated commons-io dependency version to 2.17.0.
+* Updated grpc-context version to 1.62.2.
+* Updated janino dependency version to 3.1.12.
+* Updated jetty dependency version to 9.4.56.v20240826.
+* Updated json-path dependency version to 2.9.0.
+* Updated jtidy dependency version to 1.0.5.
+* Updated log4j2 dependency version to 2.22.0.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:85]:
```suggestion
* Updated Log4j2 dependency version to 2.22.0.
```

File: RELEASE_NOTES.txt [hunk 21]
```diff
@@ -1,6 +1,177 @@
...
+* Moved Ignite sources to Java 11.
+* Optimized UUID serialization for communication messages.
+* Optimized entries expiration for persistent caches.
+* Removed deprecated TcpCommunicationSpi shared memory port configuration.
+* Removed deprecated authorization methods from Security Context.
+* Updated Jetty dependency version to 9.4.55.
+* Updated Netty dependency version to 4.1.101.Final.
+* Updated PostgreSQL JDBC dependency version to 42.7.3.
+* Updated Spring dependency version to 5.3.39.
+* Updated commons-io dependency version to 2.17.0.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:79]: Should there be back quotes ` here and below?

`commons-io`, `grpc-context`, `json-path` etc

File: RELEASE_NOTES.txt [hunk 22]
```diff
@@ -1,6 +1,177 @@
...
+* Updated grpc-context version to 1.62.2.
+* Updated janino dependency version to 3.1.12.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:81]:
```suggestion
* Updated Janino dependency version to 3.1.12.
```

File: RELEASE_NOTES.txt [hunk 23]
```diff
@@ -1,6 +1,177 @@
...
+* Updated jetty dependency version to 9.4.56.v20240826.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:82]:
```suggestion
* Updated Jetty dependency version to 9.4.56.v20240826.
```

File: RELEASE_NOTES.txt [hunk 24]
```diff
@@ -1,6 +1,177 @@
...
+* Updated json-path dependency version to 2.9.0.
+* Updated jtidy dependency version to 1.0.5.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:84]:
```suggestion
* Updated JTidy dependency version to 1.0.5.
```

File: RELEASE_NOTES.txt [hunk 25]
```diff
@@ -1,6 +1,177 @@
...
+* Updated log4j2 dependency version to 2.22.0.
+* Updated opencensus dependency version to 0.31.1.
+* Updated snappy dependency version to 1.1.10.7.
+* Updated zookeeper dependency version to 3.8.4.
+
+SQL:
+* Added ability to create user defined SQL views.
+* Added long queries tracking for DML operations.
+* Fixed AssertionError due to cache size overflow.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:93]:
```suggestion
* Fixed `AssertionError` due to cache size overflow.
```

File: RELEASE_NOTES.txt [hunk 26]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed absence of warnings for long running queries with lazy flag.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:94]:
```suggestion
* Fixed absence of warnings for long running queries with a lazy flag.
```

File: RELEASE_NOTES.txt [hunk 27]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed assertion error on extra column in H2 ResultInterface#currentRow.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:95]:
```suggestion
* Fixed assertion error on extra column in H2 `ResultInterface#currentRow`.
```

File: RELEASE_NOTES.txt [hunk 28]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed nested subquery optimization.
+* Fixed table statistics periodical update.
+
+SQL Calcite:
+* Added ability to extend SQL engine with new system functions/operators.
+* Added arithmetic overflow handling.
+* Added date formatting/parsing functions with custom format.
+* Added hint for join type.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:103]:
```suggestion
* Added a hint for join type.
```

File: RELEASE_NOTES.txt [hunk 29]
```diff
@@ -1,6 +1,177 @@
...
+* Added memory quotas control for Cursor.getAll() method.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:104]:
```suggestion
* Added memory quota control for `Cursor.getAll()` method.
```

File: RELEASE_NOTES.txt [hunk 30]
```diff
@@ -1,6 +1,177 @@
...
+* Added operations authorization.
+* Added support for SQL hints at tables.
+* Added support for bitwise aggregates (BIT_AND, BIT_OR, BIT_XOR).
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:107]:
```suggestion
* Added support for bitwise aggregates (`BIT_AND`, `BIT_OR`, `BIT_XOR`).
```

File: RELEASE_NOTES.txt [hunk 31]
```diff
@@ -1,6 +1,177 @@
...
+* Added support for bitwise operations (BITAND, BITOR, BITXOR).
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:108]:
```suggestion
* Added support for bitwise operations (`BITAND`, `BITOR`, `BITXOR`).
```

Is this a duplicate of the previous line?

File: RELEASE_NOTES.txt [hunk 32]
```diff
@@ -1,6 +1,177 @@
...
+* Added support of OR operator with dynamic parameters for index scans.
+* Allowed grouping by alias and ordinal value.
+* Fixed 'typeof' function behavior when operand throws an error (it fails now).
+* Fixed NVL/DECODE failure when used with parameters of different data types.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:112]:
```suggestion
* Fixed NVL/DECODE failure when it is used with parameters of different data types.
```

File: RELEASE_NOTES.txt [hunk 33]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed assertion error for multiline statements in thin client.
+* Fixed binary object dynamic parameter processing.
+* Fixed column nullability check for data types with precision or scale.
+* Fixed compiled expression cache key for correlated variable (to avoid cache collisions for correlated variables with different data types).
+* Fixed complex key index scans (binary objects comparison).
+* Fixed correlated nested loop LEFT join failure.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:118]: Should there be back quotes ` here and below?

`LEFT`, `TIMESTAMP`, `COUNT(DISTINCT)`

File: RELEASE_NOTES.txt [hunk 34]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed default TIMESTAMP precision.
+* Fixed error message for unsupported 'FORMAT' in 'CAST'.
+* Fixed expand star for system fields.
+* Fixed failure in COUNT(DISTINCT) operation.
+* Fixed failure on 'CREATE TABLE' DDL with Calcite engine when 'DEFAULT' column value is NULL.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:123]:
```suggestion
* Fixed failure on 'CREATE TABLE' DDL with Calcite engine when 'DEFAULT' column value is `NULL`.
```

File: RELEASE_NOTES.txt [hunk 35]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed failure on sensitive information hiding for DDL statements.
+* Fixed implicit conversion to DECIMAL for some functions.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:125]:
```suggestion
* Fixed implicit conversion to `DECIMAL` for some functions.
```

File: RELEASE_NOTES.txt [hunk 36]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed incorrect error for TIME* WITH TIME ZONE types.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:126]:
```suggestion
* Fixed incorrect error for `TIME* WITH TIME ZONE` types.
```

File: RELEASE_NOTES.txt [hunk 37]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed incorrect numeric results with operator 'IS [NOT] DISTINCT FROM'.
+* Fixed incorrect result in unions in case of different types of numeric results.
+* Fixed least restrictive type priority (for DOUBLE and FLOAT).
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:129]:
```suggestion
* Fixed least restrictive type priority (for `DOUBLE` and `FLOAT`).
```

File: RELEASE_NOTES.txt [hunk 38]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed lost limit/offset on planning.
+* Fixed lost precisions of dynamic query parameters.
+* Fixed not-null count calculation if index rebuild is in progress.
+* Fixed numeric to interval cast.
+* Fixed parsing of incorrect date literals.
+* Fixed partition reservation for index-count-scan.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:135]:
```suggestion
* Fixed partition reservation for `index-count-scan`.
```

File: RELEASE_NOTES.txt [hunk 39]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed rehashing of inputs containing null values.
+* Fixed rounding of numerics (now rounds all types of floating point numerics to nearest integer).
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:137]:
```suggestion
* Fixed rounding of numerics (now rounds all types of floating point numerics to the nearest integer).
```

File: RELEASE_NOTES.txt [hunk 40]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed sensitive information removal failure for JOINs and other SQL nodes.
+* Updated Calcite version to 1.37.
+
+Control utility:
+* Added a flag to ignore existed caches to the cache create command.
+* Added ability to view cache data in JSON format.
+* Added histogram bounds to the metric command output.
+* Added logging of the IdleVerify command.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:145]:
```suggestion
* Added logging of the `IdleVerify` command.
```

File: RELEASE_NOTES.txt [hunk 41]
```diff
@@ -1,6 +1,177 @@
...
+* Added support for the thin client connection to the control utility and deprecate legacy Binary-REST connection.
+* Fixed CDC resend of replicated entries.
+* Fixed Idle Verify hanging when invoked on an inactive cluster.
+* Fixed Idle verify check failing when debug logging level is set.
+* Fixed NPE in the cache distribution command.
+* Fixed errors output.
+* Fixed partition column output of the cache distribution command.
+* Fixed permissions required by the consistency repair command.
+* Removed ADMIN_OPS permissions check for cache operations.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:154]:
```suggestion
* Removed `ADMIN_OPS` permissions check for cache operations.
```

File: RELEASE_NOTES.txt [hunk 42]
```diff
@@ -1,6 +1,177 @@
...
+
+Java thin client:
+* Added ClientClusterGroup support for Service Awareness.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:157]:
```suggestion
* Added `ClientClusterGroup` support for Service Awareness.
```

File: RELEASE_NOTES.txt [hunk 43]
```diff
@@ -1,6 +1,177 @@
...
+* Added events for client start and stop.
+* Added invoke/invokeAll operations.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:159]:
```suggestion
* Added `invoke`/`invokeAll` operations.
```

File: RELEASE_NOTES.txt [hunk 44]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed partition awareness after cluster restart with the same topology version.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:160]:
```suggestion
* Fixed partition awareness after a cluster restart with the same topology version.
```

File: RELEASE_NOTES.txt [hunk 45]
```diff
@@ -1,6 +1,177 @@
...
+* Fixed client init hanging when cluster discovery is enabled but returned addresses are unreachable (in another subnet).
+* Fixed client-connector threads blocking by transactional operations.
+* Fixed orphaned handles after binary object detach operation.
+* Fixed partition awareness with node filter.
+* Fixed stale client cluster group data after cluster restart.
```
Comment (SofiaSedova) [inline RELEASE_NOTES.txt:165]:
```suggestion
* Fixed stale client cluster group data after cluster restart.
```
```suggestion
* Fixed stale client cluster group data after a cluster restart.
```

**Pull request merged**